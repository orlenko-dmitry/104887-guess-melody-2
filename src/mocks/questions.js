const questions = [
  {
    "type": `artist`,
    "song": {
      "artist": `The Beatles`,
      "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/6/6d/National_anthem_of_Ukraine%2C_instrumental.oga/National_anthem_of_Ukraine%2C_instrumental.oga.mp3`
    },
    "answers": [
      {
        "id": 1,
        "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
        "artist": `The Beatles`
      },
      {
        "id": 2,
        "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`,
        "artist": `Oasis`
      },
      {
        "id": 3,
        "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg`,
        "artist": `The Kinks`
      }
    ]
  },
  {
    "type": `genre`,
    "genre": `reggae`,
    "answers": [
      {
        "id": 1,
        "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/6/6d/National_anthem_of_Ukraine%2C_instrumental.oga/National_anthem_of_Ukraine%2C_instrumental.oga.mp3`,
        "genre": `reggae`
      },
      {
        "id": 2,
        "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b4/United_States_Navy_Band_-_O_Canada.ogg/United_States_Navy_Band_-_O_Canada.ogg.mp3`,
        "genre": `electronic`
      },
      {
        "id": 3,
        "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/3/30/La_Marseillaise.ogg/La_Marseillaise.ogg.mp3`,
        "genre": `country`
      },
      {
        "id": 4,
        "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/c/c8/Marcha_Real-Royal_March_by_US_Navy_Band.ogg/Marcha_Real-Royal_March_by_US_Navy_Band.ogg.mp3`,
        "genre": `reggae`
      }
    ]
  }
];

export default questions;
