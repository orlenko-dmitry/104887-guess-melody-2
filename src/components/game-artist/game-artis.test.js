import React from 'react';
import renderer from 'react-test-renderer';

import GameArtist from './game-artist.jsx';
import questions from '../../mocks/questions.js';

it(`GameArtist renders correctly`, () => {
  const tree = renderer
    .create(
        <GameArtist
          gameData={questions[0]}
          onNextScreenClick={() => {}}
          onSetAnswerClik={() => {}}
        />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
