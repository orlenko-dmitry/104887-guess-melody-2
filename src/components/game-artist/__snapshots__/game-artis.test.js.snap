// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameArtist renders correctly 1`] = `
<section
  class="game game--artist"
>
  <header
    class="game__header"
  >
    <a
      class="game__back"
      href="#"
    >
      <span
        class="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        class="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      class="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="timer__line"
        cx="390"
        cy="390"
        r="370"
        style="filter: url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center;"
      />
    </svg>
    <div
      class="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        class="timer__mins"
      >
        05
      </span>
      <span
        class="timer__dots"
      >
        :
      </span>
      <span
        class="timer__secs"
      >
        00
      </span>
    </div>
    <div
      class="game__mistakes"
    >
      <div
        class="wrong"
      />
      <div
        class="wrong"
      />
      <div
        class="wrong"
      />
    </div>
  </header>
  <section
    class="game__screen"
  >
    <h2
      class="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      class="game__track"
    >
      <div
        class="track"
      >
        <button
          class="track__button track__button--play"
          data-testid="play-btn"
          disabled=""
          type="button"
        />
        <div
          class="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/6/6d/National_anthem_of_Ukraine%2C_instrumental.oga/National_anthem_of_Ukraine%2C_instrumental.oga.mp3"
          />
        </div>
      </div>
    </div>
    <form
      class="game__artist"
    >
      <div
        class="artist"
        data-testid="artist-0"
      >
        <input
          class="artist__input visually-hidden"
          id="1"
          name="answer"
          type="radio"
          value="The Beatles"
        />
        <label
          class="artist__name"
          for="1"
        >
          <img
            alt="Пелагея"
            class="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
          />
          The Beatles
        </label>
      </div>
      <div
        class="artist"
        data-testid="artist-1"
      >
        <input
          class="artist__input visually-hidden"
          id="2"
          name="answer"
          type="radio"
          value="Oasis"
        />
        <label
          class="artist__name"
          for="2"
        >
          <img
            alt="Пелагея"
            class="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg"
          />
          Oasis
        </label>
      </div>
      <div
        class="artist"
        data-testid="artist-2"
      >
        <input
          class="artist__input visually-hidden"
          id="3"
          name="answer"
          type="radio"
          value="The Kinks"
        />
        <label
          class="artist__name"
          for="3"
        >
          <img
            alt="Пелагея"
            class="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg"
          />
          The Kinks
        </label>
      </div>
    </form>
  </section>
</section>
`;
